packages/@aws-cdk/aws-lambda/lib/function-url.ts

```diff
@@ -0,0 +1,225 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Duration, IResource, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IAlias } from './alias';
+import { IFunction } from './function-base';
+import { IVersion } from './lambda-version';
+import { CfnUrl } from './lambda.generated';
+
+
+/**
+ * The auth types for a function url
+ */
+export enum FunctionUrlAuthType {
+  /**
+   * Restrict access to authenticated IAM users only
+   */
+  AWS_IAM = 'AWS_IAM',
+
+  /**
+   * Bypass IAM authentication to create a public endpoint
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * All http request methods
+ */
+export enum HttpMethods {
+  /**
+   * The GET method requests a representation of the specified resource.
+   */
+  GET = 'GET',
+  /**
+   * The PUT method replaces all current representations of the target resource with the request payload.
+   */
+  PUT = 'PUT',
+  /**
+   * The HEAD method asks for a response identical to that of a GET request, but without the response body.
+   */
+  HEAD = 'HEAD',
+  /**
+   * The POST method is used to submit an entity to the specified resource, often causing a change in state or side effects on the server.
+   */
+  POST = 'POST',
+  /**
+   * The DELETE method deletes the specified resource.
+   */
+  DELETE = 'DELETE',
+  /**
+   * The PATCH method applies partial modifications to a resource.
+   */
+  PATCH = 'PATCH',
+  /**
+   * The OPTIONS method describes the communication options for the target resource.
+   */
+  OPTIONS = 'OPTIONS',
+  /**
+   * The wildcard entry to allow all methods.
+   */
+  ALL = '*',
+}
+
+/**
+ * Specifies a cross-origin access property for a function URL
+ */
+export interface FunctionUrlCorsOptions {
+  /**
+   * Whether to allow cookies or other credentials in requests to your function URL.
+   *
+   * @default false
+   */
+  readonly allowCredentials?: boolean;
+
+  /**
+   * Headers that are specified in the Access-Control-Request-Headers header.
+   *
+   * @default - No headers allowed.
+   */
+  readonly allowedHeaders?: string[];
+
+  /**
+   * An HTTP method that you allow the origin to execute.
+   *
+   * @default - No methods allowed.
+   */
+  readonly allowedMethods?: HttpMethods[];
```

Comment:
I think I'd want to default `allowedMethods` to `*`. That seems like an easy UX optimization with minimal security repercussions.